### PR TITLE
test/e2e: dedup Before/AfterEach nodes

### PR DESCRIPTION
**Pull request by Luap99**
There is no reason to define the same code every time in each file, just use global nodes. This diff should speak for itself.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
Nice simplification
LGTM
@edsantiago PTAL

**Comment by TomSweeneyRedHat** [comment]
Big Gold Star for this one @Luap99 :star2: 
@edsantiago PTAL
LGTM, but let's let Ed merge this

File: test/e2e/common_test.go [hunk 1]
```diff
@@ -102,6 +102,25 @@ func TestLibpod(t *testing.T) {
...
+
+	_ = BeforeEach(func() {
+		tempdir, err = CreateTempDirInTempDir()
+		Expect(err).ToNot(HaveOccurred())
+		podmanTest = PodmanTestCreate(tempdir)
+		podmanTest.Setup()
+	})
+
+	_ = AfterEach(func() {
+		podmanTest.Cleanup()
```
Comment (edsantiago) [inline test/e2e/common_test.go:118]: In the `volume_*` tests (far below at bottom), the cleanup function is `CleanupVolume()`. Your PR loses that. I'm not sure how best to address it, it depends on the ordering of `AfterEach()`.

**Review by edsantiago — COMMENTED** [review]
I really love this, thank you! I'm concerned about the ordering of `AfterEach()`. (`BeforeEach()`, it seems clear that `common` runs _before_ the test-specific one). Specifically, in tests where cleanup involves `Unsetenv("CONTAINERS_CONF")`, can you confirm that the ordering of `Unsetenv()`-`Cleanup()` is safe, or does not matter? Also, the `Volumes` thing below.

Thank you for taking this on!

File: test/e2e/common_test.go (hunk 1 above)
Comment (Luap99) [reply test/e2e/common_test.go:118]: very good eyes, I will fix this.

**Comment by Luap99** [comment]
>  Specifically, in tests where cleanup involves Unsetenv("CONTAINERS_CONF"), can you confirm that the ordering of Unsetenv()-Cleanup() is safe, or does not matter?

It most cases it does not matter, however while working on ginkgov2 I found a case where it matters. When tests use an invalid CONTAINERS_CONF file which contains invalid options or maybe does not even exists podman will just error out. Thus the Cleanup() could fail in this case. Therefore the correct order is to always unsetenv before Cleanup().

**Comment by Luap99** [comment]
given how many tests use CONTAINERS_CONF I could juts move it in the main AfterEach() block and let all test unset it. There shouldn't be a problem with that and it prevents other from forgetting to unset it.

**Comment by edsantiago** [comment]
Seems fair. I did a very quick check, and don't see any places where `Cleanup()` would require the test's `CONTAINERS_CONF`, but I don't fully trust myself to be certain. This is tricky.

**Comment by edsantiago** [comment]
This is a lot cleaner and looks safer, thank you! Thanks for documenting the execution order and for the link. A few typos in the commit message, not worth re-pushing for, but if a re-push is needed for anything else, please take a quick edit pass there.

File: test/e2e/image_scp_test.go [hunk 2]
```diff
@@ -13,41 +13,10 @@ import (
...
-		if ConfPath.IsSet {
-			os.Setenv("CONTAINERS_CONF", ConfPath.Value)
-		} else {
-			os.Unsetenv("CONTAINERS_CONF")
-		}
-		f := CurrentSpecReport()
-		processTestResult(f)
-
+		// make sure connections are not written to real user config on host
+		os.Setenv("CONTAINERS_CONF", filepath.Join(podmanTest.TempDir, "containersconf"))
```
Comment (edsantiago) [inline test/e2e/image_scp_test.go:19]: Looks like you need to create this.

**Comment by edsantiago** [comment]
>Ginkgo orders the BeforeEach and AfterEach nodes. They will be executed
>from the outer-most defnied to inner-most. This mean out global

"defined"; "our"

>BeforeEAch is always first. Only then the inner one (in the Describe() funtion

"Each"; "function"

File: test/e2e/image_scp_test.go (hunk 2 above)
Comment (Luap99) [reply test/e2e/image_scp_test.go:19]: Guess I deserve that for not running it locally first.

**Pull request merged**